src/network/networkd-network.c

```diff
@@ -342,9 +343,9 @@ int network_load_one(Manager *manager, OrderedHashmap **networks, const char *fi
         if (!fname)
                 return log_oom();
 
-        name = strdup(basename(filename));
-        if (!name)
-                return log_oom();
+        r = path_extract_filename(filename, &name);
+        if (r < 0)
+                return log_warning_errno(r, "Couldn't determine file name for \"%s\": %m", filename);
```

Comment:
I'd like to make the all log messages consistent:
`"Failed to extract file name of '%s': %m"`